db/migrations/00023_create_task_key_map.go

```diff
@@ -0,0 +1,44 @@
+package migrations
+
+import (
+	"database/sql"
+	"github.com/iReflect/reflect-app/db/base/models"
+	"github.com/jinzhu/gorm"
+	"github.com/pressly/goose"
+)
+
+func init() {
+	goose.AddMigration(Up00023, Down00023)
+}
+
+// Up00023 ...
+func Up00023(tx *sql.Tx) error {
+	// This code is executed when the migration is applied.
+	gormdb, err := gorm.Open("postgres", interface{}(tx).(gorm.SQLCommon))
+	if err != nil {
+		return err
+	}
+
+	gormdb.AutoMigrate(&models.TaskKeyMap{})
+	gormdb.Model(&models.TaskKeyMap{}).AddForeignKey("task_id", "tasks(id)", "RESTRICT", "RESTRICT")
+	gormdb.Model(&models.TaskKeyMap{}).AddUniqueIndex("unique_task_id_key", "task_id", "key")
+	gormdb.Model(&models.TaskKeyMap{}).AddIndex("idx_task_key", "key")
```

Comment:
We can make `key` the leading column in the index above to skip this index.

Reply:
done